### ISSUE TITLE
Tabular: Early stopping is too aggressive

### ISSUE THREAD

**Issue by jeremyholcombe** [awslabs/autogluon#714]
There seems to be some kind of global early stopping setting that I'm having difficulty disabling. Is there early stopping setting in `task.fit` that is agnostic of the algorithm being used? **If so, is there any way I can disable it so I can run the full iterations specified?**

I am trying to train two CatBoost configurations with 5 bagging folds and 1 stacked ensemble _without_ time limits. I am able to train a full 3,000 iterations using a Catboost classifier (not from Autogluon) with the same configuration, but when I employ this config in Autogluon, it early stops around 1,000 iterations consistently. Furthermore, the validation loss does not increase at any point, so I also am not sure why the early stopping is being triggered (and it does not seem to be Catboost triggering it). 

Here are the Catboost configurations:
```python
cat1_params = {
    'iterations': 3000,
  'learning_rate': 0.4,
  'depth': 7,
  'l2_leaf_reg': 3,
  'model_size_reg': 0.5,
  'use_best_model': True,
  'od_type': 'Iter',
  'od_wait': 2000,
  'task_type': 'GPU'
}
 
cat2_params = {
  'iterations': 3000,
  'learning_rate': 0.3,
  'depth': 6,
  'l2_leaf_reg': 3,
  'use_best_model': True,
  'od_type': 'Iter',
  'od_wait': 2000,
  'task_type': 'GPU'
}
```

And here is the `task.fit` configuration:
```python
predictor = task.fit(train_data=train_data, label=label_column,
                     output_directory=dbfs_dir, eval_metric='log_loss', verbosity=3,
                     num_bagging_folds=5, stack_ensemble_levels=1,
                     hyperparameters = { 'CAT':[cat1_params, cat2_params]},
                     presets='ignore_text')
```

As you can see from this image here, both training and validation loss decreases monotonically throughout the entire run, but something is triggering early stopping before it is able to run the 3000 iterations specified.

![image](https://user-images.githubusercontent.com/19765330/96516162-41660c80-122c-11eb-9774-9bbd8db0bcbb.png)

**Comment by Innixma** [comment]
Thanks for submitting an issue!

I suspect that the model actually did have the validation loss increase which triggered the early stopping. For large enough datasets the early stopping rounds is 10, so you would need to set verbosity=4 to see the per-iteration losses rather than every 20 iterations to check for sure.

In 0.0.14, unfortunately there is no option to disable or change the early stopping rounds of the CatBoost and LightGBM models.

It is currently set by this code:

```
        if X_val is not None:
            X_val = self.preprocess(X_val)
            X_val = Pool(data=X_val, label=y_val, cat_features=cat_features)
            eval_set = X_val
            if num_rows_train <= 10000:
                modifier = 1
            else:
                modifier = 10000/num_rows_train
            early_stopping_rounds = max(round(modifier*150), 10)
            num_sample_iter_max = max(round(modifier*50), 2)
        else:
            eval_set = None
            early_stopping_rounds = None
            num_sample_iter_max = 50
```

To get around this, you can edit the code in `autogluon/utils/tabular/ml/models/catboost/catboost_model.py`

I will plan to make this overridable in v0.1 release. This way you should get roughly identical results training CatBoost alone compared to with AutoGluon, and you will be able to extend/turn off early stopping.

**Comment by AhangarAamir** [comment]
How to turn off early stopping.
